packages/internal/src/storage-test-utils.js

```diff
@@ -1,11 +1,67 @@
 // @ts-check
 import { Far } from '@endo/far';
-import { makeMarshal } from '@endo/marshal';
+import { makeMarshal, Remotable } from '@endo/marshal';
 import { makeChainStorageRoot } from './lib-chainStorage.js';
 import { bindAllMethods } from './method-tools.js';
 
 const { Fail, quote: q } = assert;
 
+/**
+ * A convertSlotToVal function that produces basic Remotables
+ *
+ * @param {string} _slotId
+ * @param {string} iface
+ */
+export const slotToRemotable = (_slotId, iface = 'Alleged: Remotable') =>
+  Remotable(iface);
+
+/**
+ * A basic marshaller whose unserializer produces Remotables. It can
+ * only serialize plain data, not Remotables.
+ */
+export const defaultMarshaller = makeMarshal(undefined, slotToRemotable, {
+  serializeBodyFormat: 'smallcaps',
+});
+
+/**
+ * A deserializer which produces slot strings instead of Remotables,
+ * so if `a = Far('iface')`, and serializing `{ a }` into `capData`
+ * assigned it slot `board123`, then `slotStringUnserialize(capData)`
+ * would produce `{ a: 'board123' }`.
+ *
+ * This may be useful for display purposes.
+ *
+ * One limitation: serialized data that contains a particular unusual
+ * string will be unserialized into a BigInt.
```

Comment:
Also, we rely on the absence of promises here too.

Reply:
You're right, I'll add the disclaimer.

The code that uses this is in lib-wallet.js, like https://github.com/Agoric/agoric-sdk/blob/720a16fd76a528868364bf107153c87e8a00e54c/packages/wallet/api/src/lib-wallet.js#L383 (where `fillInSlots` is really `slotStringUnserialize`). I don't know what/why that code does, but I'm guessing it's trying to produce a human-readable record of data from some capdata (note `display` in the name), and an object tree with slot strings is going to be more readable than `capData.body`.

I believe all of this data is coming from fairly static storage (RPC calls pulling from chainStorage), so it will never contain promises (there would be no pathway to subscribe for changes, nor a way to unsubscribe afterwards).

Proper UI code would need to have an interactive-ish petname system to do something meaningful with the board IDs / slots, so I expect that the deeper question is how our UI works, and under what circumstances this function is being used to display data to users.

Reply:
`lib-wallet.js` is not connected to chainStorage. It's only used in the ag-solo wallet, and not in production, IIUC. I forget why we needed to hydrate/dehydrate... something about browser localStorage maybe?

I think @michaelfig knows.

Reply:
hydrate/dehydrate was an invention to map arbitrary ocaps to petnames (or "unknownXXX") for display in the UI.  Since the time it was invented, we've come to agree that `lib-wallet`'s implementation of a petname system is subpar, as it is inconsistent between ocap types, and flawed in using those petnames as strongnames when communicating with the wallet UI.

So, please don't break it unnecessarily as it works better than nothing, but we will likely have to retire ag-solo and its ad hoc interactions with the chain in favour of an Endo app that subsumes its functionality.

Reply:
"strongname"?

Reply:
> "strongname"?

Maybe I mean "clist entry".

Reply:
Cool!